src/proficiency.cpp

```diff
@@ -218,6 +228,30 @@ void proficiency_set::remove( const proficiency_id &lost )
     }
 }
 
+void proficiency_set::direct_learn( const proficiency_id &learned )
+{
+    // Player might be learning proficiency
+    for( std::vector<learning_proficiency>::iterator it = learning.begin(); it != learning.end(); ) {
+        if( it->id == learned ) {
+            it = learning.erase( it );
+        } else {
+            ++it;
+        }
+    }
+
```

Comment:
You might want to use the `erase( remove_if( ... ) )` idiom here (although not if the order of the vector is important).

Reply:
The order might be important, I copied this part from `src/proficiency.cpp:165`, I didn't search in detail on why the code author wrote it like this but there is probably a reason.